proseco/gui.py

```diff
@@ -0,0 +1,358 @@
+
+    q = Quat((ra, dec, roll))
+    yag_deg, zag_deg = radec2yagzag(cone_stars['RA_PMCORR'], cone_stars['DEC_PMCORR'], q)
+    row, col = chandra_aca.yagzag_to_pixels(yag_deg * 3600,
+                                            zag_deg * 3600, allow_bad=True)
+    # update these for every new roll
+    cone_stars['yang'] = yag_deg * 3600
+    cone_stars['zang'] = zag_deg * 3600
+    cone_stars['row'] = row
+    cone_stars['col'] = col
+
+    # none of these appear stage dependent, but they could be type (guide/acq) dependent
+    chip_edge_dist, fov_edge_dist, offchip, outofbounds = check_off_chips(cone_stars, opt)
+    #cone_stars['offchip_{}'.format(stype)] = offchip
+    #cone_stars['outofbounds_{}'.format(stype)] = outofbounds
+    cone_stars['chip_edge_dist_{}'.format(stype)] = chip_edge_dist
+    cone_stars['fov_edge_dist_{}'.format(stype)] = fov_edge_dist
+
+    bad_mag_error = cone_stars['MAG_ACA_ERR'] > opt['Inertial']['MagErrorTol']
+    #cone_stars['bad_mag_error_{}'.format(stype)] = bad_mag_error
+
+    bad_pos_error = cone_stars['POS_ERR'] > opt['Inertial']['PosErrorTol']
+    #cone_stars['bad_pos_error_{}'.format(stype)] = bad_pos_error
+
+    bad_aspq1 = ((cone_stars['ASPQ1'] > np.max(opt['Inertial']['ASPQ1Lim']))
+                  | (cone_stars['ASPQ1'] < np.min(opt['Inertial']['ASPQ1Lim'])))
+    #cone_stars['bad_aspq1_{}'.format(stype)] = bad_aspq1
+
+    bad_aspq2 = ((cone_stars['ASPQ2'] > np.max(opt['Inertial']['ASPQ2Lim']))
+                  | (cone_stars['ASPQ2'] < np.min(opt['Inertial']['ASPQ2Lim'])))
+    #cone_stars['bad_aspq2_{}'.format(stype)] = bad_aspq2
+
+    bad_aspq3 = ((cone_stars['ASPQ3'] > np.max(opt['Inertial']['ASPQ3Lim']))
+                  | (cone_stars['ASPQ3'] < np.min(opt['Inertial']['ASPQ3Lim'])))
+    #cone_stars['bad_aspq3_{}'.format(stype)] = bad_aspq3
+
+
+    nonstellar = cone_stars['CLASS'] != 0
+    #cone_stars['nonstellar'] = nonstellar
+
+    not_bad = (~offchip & ~outofbounds & ~bad_mag_error & ~bad_pos_error
+                & ~nonstellar & ~bad_aspq1 & ~bad_aspq2 & ~bad_aspq3)
+
+    # Set some column defaults that will be updated in check_stage
+    cone_stars['{}_stage'.format(stype)] = -1
+    ncand = opt['Select']['NMaxSelect'] + opt['Select']['nSurplus']
+    for idx, stage_char in enumerate(STAR_CHAR[stype], 1):
+        # Save the type in the characteristics
+        stage_char['Type'] = stype
+        stage_char['Stage'] = idx
+        if np.count_nonzero(cone_stars['{}_stage'.format(stype)] != -1) < ncand:
+            stage  = check_stage(cone_stars,
+                                 not_bad & ~(cone_stars['{}_stage'.format(stype)] != -1),
+                                 stage_char, "{}_{}".format(stype, str(idx)))
+            cone_stars['{}_stage'.format(stype)][stage] = idx
+    selected = cone_stars[cone_stars['{}_stage'.format(stype)] != -1]
+    return selected
+
+
+
+def select_guide_stars(ra, dec, roll, dither=8, n=5, cone_stars=None, date=None):
```

Comment:
Just use `stars` everywhere instead of `cone_stars`.  Then in the line below you can filter the stars to be in the ACA FOV box (with margin as necessary) to avoid useless stars in the list downstream.  This is consistent with usage in `acq.py`.  In fact it might make sense to factor out `acq.get_stars()` to a `utils.py` module and then use it here.

Reply:
"In fact it might make sense to factor out acq.get_stars() to a utils.py module and then use it here."

I was going to open that as an issue/suggestion that they share their star fetching etc, but then I figured you'd give me a hard time for premature optimization.